waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -923,7 +923,26 @@
            :resources {:limits {:memory memory}
                        :requests {:cpu cpu :memory memory}}
            :volumeMounts [{:mountPath "/srv/www"
-                           :name "user-home"}]})))))
+                           :name "user-home"}]}))
+
+      ;; Optional envoy sidecar container
+      has-reverse-proxy?
+      (update-in
+        [:spec :template :spec :container]
+        conj
+        (let [cmd ["/bin/sh" "/opt/waiter/envoy/bin/envoy-start"]]
+             {:command cmd
+              :env (into [{:name "SERVICE-PORT" :value service-port}
+                          {:name "PORT0" :value port0}]
+                         (concat
+                           (for [[k v] base-env]
+                                {:name k :value v})))
+              :image "twosigma/waiter-envoy"
```

Comment:
This needs to be parameterized so we can set it in `config.edn`, similar to the file server's image.